# TITANIC DATASET

# Look at Data

https://www.dataquest.io/blog/pandas-cheat-sheet/

https://pandas.pydata.org/docs/getting_started/intro_tutorials/02_read_write.html

https://www.codegrepper.com/code-examples/delphi/how+to+drop+only+nan+values+in+pandas

https://chrisalbon.com/python/data_wrangling/pandas_dropping_column_and_rows/

In [21]:
# Reading data in Pandas

import pandas as pd # import Pandas

df = pd.read_csv("titanic_data.csv") # Read data file
df.head()
df[0:5] # Display data

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [22]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [23]:
df.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

In [24]:
df.tail()

PassengerId  Survived  Pclass                                      Name  \
886          887         0       2                     Montvila, Rev. Juozas   
887          888         1       1              Graham, Miss. Margaret Edith   
888          889         0       3  Johnston, Miss. Catherine Helen "Carrie"   
889          890         1       1                     Behr, Mr. Karl Howell   
890          891         0       3                       Dooley, Mr. Patrick   

        Sex   Age  SibSp  Parch      Ticket   Fare Cabin Embarked  
886    male  27.0      0      0      211536  13.00   NaN        S  
887  female  19.0      0      0      112053  30.00   B42        S  
888  female   NaN      1      2  W./C. 6607  23.45   NaN        S  
889    male  26.0      0      0      111369  30.00  C148        C  
890    male  32.0      0      0      370376   7.75   NaN        Q

In [25]:
df.dtypes

PassengerId      int64
Survived         int64
Pclass           int64
Name            object
Sex             object
Age            float64
SibSp            int64
Parch            int64
Ticket          object
Fare           float64
Cabin           object
Embarked        object
dtype: object

In [26]:
df[["Age","Sex"]]

Age     Sex
0    22.0    male
1    38.0  female
2    26.0  female
3    35.0  female
4    35.0    male
..    ...     ...
886  27.0    male
887  19.0  female
888   NaN  female
889  26.0    male
890  32.0    male

[891 rows x 2 columns]

In [27]:
df[df['Age']>35]

PassengerId  Survived  Pclass  \
1              2         1       1   
6              7         0       1   
11            12         1       1   
13            14         0       3   
15            16         1       2   
..           ...       ...     ...   
865          866         1       2   
871          872         1       1   
873          874         0       3   
879          880         1       1   
885          886         0       3   

                                                  Name     Sex   Age  SibSp  \
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
6                              McCarthy, Mr. Timothy J    male  54.0      0   
11                            Bonnell, Miss. Elizabeth  female  58.0      0   
13                         Andersson, Mr. Anders Johan    male  39.0      1   
15                    Hewlett, Mrs. (Mary D Kingcome)   female  55.0      0   
..                                                 ...     ...   ...    ...   
865                           Bystrom, Mrs. (Karolina)  female  42.0      0   
871   Beckwith, Mrs. Richard Leonard (Sallie Monypeny)  female  47.0      1   
873                        Vander Cruyssen, Mr. Victor    male  47.0      0   
879      Potter, Mrs. Thomas Jr (Lily Alexenia Wilson)  female  56.0      0   
885               Rice, Mrs. William (Margaret Norton)  female  39.0      0   

     Parch    Ticket     Fare Cabin Embarked  
1        0  PC 17599  71.2833   C85        C  
6        0     17463  51.8625   E46        S  
11       0    113783  26.5500  C103        S  
13       5    347082  31.2750   NaN        S  
15       0    248706  16.0000   NaN        S  
..     ...       ...      ...   ...      ...  
865      0    236852  13.0000   NaN        S  
871      1     11751  52.5542   D35        S  
873      0    345765   9.0000   NaN        S  
879      1     11767  83.1583   C50        C  
885      5    382652  29.1250   NaN        Q  

[217 rows x 12 columns]

In [28]:
df[df['Age']>35].shape

(217, 12)

In [29]:
df[df['Age']>35].dtypes

PassengerId      int64
Survived         int64
Pclass           int64
Name            object
Sex             object
Age            float64
SibSp            int64
Parch            int64
Ticket          object
Fare           float64
Cabin           object
Embarked        object
dtype: object

In [30]:
df["Age"]

0      22.0
1      38.0
2      26.0
3      35.0
4      35.0
       ... 
886    27.0
887    19.0
888     NaN
889    26.0
890    32.0
Name: Age, Length: 891, dtype: float64

**csv to excel**

In [31]:
import pandas as pd

read_file = pd.read_csv (r"titanic_data.csv")
read_file.to_excel (r'titanic_data.xlsx', index = None, header=True)

In [32]:
df.to_excel('titanic.xlsx', sheet_name='passengers', index=False)

In [33]:
# Reading the csv file 
df_new = pd.read_csv('titanic_data.csv') 
  
# saving xlsx file 
GFG = pd.ExcelWriter('titanic_data.xlsx') 
df_new.to_excel(GFG, index = False) 

In [34]:
pd.read_csv('titanic_data.csv').to_excel('titanic_data.xlsx')

**I’m interested in the Titanic passengers from cabin class 2 and 3.**

In [78]:
df[df['Pclass']==2, df['Pclass']==3]

TypeError: '(0      False
1      False
2      False
3      False
4      False
       ...  
886     True
887    False
888    False
889    False
890    False
Name: Pclass, Length: 891, dtype: bool, 0       True
1      False
2       True
3      False
4       True
       ...  
886    False
887    False
888     True
889    False
890     True
Name: Pclass, Length: 891, dtype: bool)' is an invalid key

I want to work with passenger data for which the age is known.

In [36]:
 df[df['Age'].notna()]

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
885          886         0       3   
886          887         0       2   
887          888         1       1   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
885               Rice, Mrs. William (Margaret Norton)  female  39.0      0   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
885      5            382652  29.1250   NaN        Q  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[714 rows x 12 columns]

I’m interested in the names of the passengers older than 35 years.

In [37]:
 df[df['Age']>35]

PassengerId  Survived  Pclass  \
1              2         1       1   
6              7         0       1   
11            12         1       1   
13            14         0       3   
15            16         1       2   
..           ...       ...     ...   
865          866         1       2   
871          872         1       1   
873          874         0       3   
879          880         1       1   
885          886         0       3   

                                                  Name     Sex   Age  SibSp  \
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
6                              McCarthy, Mr. Timothy J    male  54.0      0   
11                            Bonnell, Miss. Elizabeth  female  58.0      0   
13                         Andersson, Mr. Anders Johan    male  39.0      1   
15                    Hewlett, Mrs. (Mary D Kingcome)   female  55.0      0   
..                                                 ...     ...   ...    ...   
865                           Bystrom, Mrs. (Karolina)  female  42.0      0   
871   Beckwith, Mrs. Richard Leonard (Sallie Monypeny)  female  47.0      1   
873                        Vander Cruyssen, Mr. Victor    male  47.0      0   
879      Potter, Mrs. Thomas Jr (Lily Alexenia Wilson)  female  56.0      0   
885               Rice, Mrs. William (Margaret Norton)  female  39.0      0   

     Parch    Ticket     Fare Cabin Embarked  
1        0  PC 17599  71.2833   C85        C  
6        0     17463  51.8625   E46        S  
11       0    113783  26.5500  C103        S  
13       5    347082  31.2750   NaN        S  
15       0    248706  16.0000   NaN        S  
..     ...       ...      ...   ...      ...  
865      0    236852  13.0000   NaN        S  
871      1     11751  52.5542   D35        S  
873      0    345765   9.0000   NaN        S  
879      1     11767  83.1583   C50        C  
885      5    382652  29.1250   NaN        Q  

[217 rows x 12 columns]

In [38]:
 k=df[df['Age']>35]

In [39]:
df[df['Age']>35]["Name"]

1      Cumings, Mrs. John Bradley (Florence Briggs Th...
6                                McCarthy, Mr. Timothy J
11                              Bonnell, Miss. Elizabeth
13                           Andersson, Mr. Anders Johan
15                      Hewlett, Mrs. (Mary D Kingcome) 
                             ...                        
865                             Bystrom, Mrs. (Karolina)
871     Beckwith, Mrs. Richard Leonard (Sallie Monypeny)
873                          Vander Cruyssen, Mr. Victor
879        Potter, Mrs. Thomas Jr (Lily Alexenia Wilson)
885                 Rice, Mrs. William (Margaret Norton)
Name: Name, Length: 217, dtype: object

What are the name of columns ?

The **columns** function gives the name of the columns in your data file.

In [172]:
column=list(df.columns)
print(column)

['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']


What is the average age of the Titanic passengers

In [174]:
df['Age'].mean()

29.69911764705882

Approx. 30 years

What is the maximum and minimum age of the Titanic passengers?

In [177]:
df['Age'].max()

80.0

In [178]:
df['Age'].min()

0.42

Did the minimum age person survive?

In [180]:
df[(df['Age'] ==0.42) & (df['Survived'] >0)]

PassengerId  Survived  Pclass                             Name   Sex  \
803          804         1       3  Thomas, Master. Assad Alexander  male   

      Age  SibSp  Parch Ticket    Fare Cabin Embarked  
803  0.42      0      1   2625  8.5167   NaN        C

yes thats great new

What is the median age and ticket fare price of the Titanic passengers?

In [183]:
(df['Age']).median()

28.0

In [184]:
(df['Fare']).median()

14.4542

In [185]:
df[["Fare", "Age"]].median()

Fare    14.4542
Age     28.0000
dtype: float64

What is the average age for male versus female Titanic passengers?

In [42]:
df.groupby('Sex')['Age'].mean()

Sex
female    27.915709
male      30.726645
Name: Age, dtype: float64

group by sex and age

In [189]:
df[['Sex','Age']]

Sex   Age
0      male  22.0
1    female  38.0
2    female  26.0
3    female  35.0
4      male  35.0
..      ...   ...
886    male  27.0
887  female  19.0
888  female   NaN
889    male  26.0
890    male  32.0

[891 rows x 2 columns]

 Another way to find the # of passengers older than 35 years ?

In [190]:
df[df['Age']>35].count()

PassengerId    217
Survived       217
Pclass         217
Name           217
Sex            217
Age            217
SibSp          217
Parch          217
Ticket         217
Fare           217
Cabin           95
Embarked       215
dtype: int64

In [191]:
age_above_35=df[df['Age']>35]
age_above_35.head()

PassengerId  Survived  Pclass  \
1             2         1       1   
6             7         0       1   
11           12         1       1   
13           14         0       3   
15           16         1       2   

                                                 Name     Sex   Age  SibSp  \
1   Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
6                             McCarthy, Mr. Timothy J    male  54.0      0   
11                           Bonnell, Miss. Elizabeth  female  58.0      0   
13                        Andersson, Mr. Anders Johan    male  39.0      1   
15                   Hewlett, Mrs. (Mary D Kingcome)   female  55.0      0   

    Parch    Ticket     Fare Cabin Embarked  
1       0  PC 17599  71.2833   C85        C  
6       0     17463  51.8625   E46        S  
11      0    113783  26.5500  C103        S  
13      5    347082  31.2750   NaN        S  
15      0    248706  16.0000   NaN        S

In [192]:
age_above_35.shape

(217, 12)

I am interested in the Titanic passengers from cabin class 2 and 3.

In [44]:
df['Pclass']

0      3
1      1
2      3
3      1
4      3
      ..
886    2
887    1
888    3
889    1
890    3
Name: Pclass, Length: 891, dtype: int64

In [196]:
df[df['Pclass'] ==1]

PassengerId  Survived  Pclass  \
1              2         1       1   
3              4         1       1   
6              7         0       1   
11            12         1       1   
23            24         1       1   
..           ...       ...     ...   
871          872         1       1   
872          873         0       1   
879          880         1       1   
887          888         1       1   
889          890         1       1   

                                                  Name     Sex   Age  SibSp  \
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
6                              McCarthy, Mr. Timothy J    male  54.0      0   
11                            Bonnell, Miss. Elizabeth  female  58.0      0   
23                        Sloper, Mr. William Thompson    male  28.0      0   
..                                                 ...     ...   ...    ...   
871   Beckwith, Mrs. Richard Leonard (Sallie Monypeny)  female  47.0      1   
872                           Carlsson, Mr. Frans Olof    male  33.0      0   
879      Potter, Mrs. Thomas Jr (Lily Alexenia Wilson)  female  56.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
889                              Behr, Mr. Karl Howell    male  26.0      0   

     Parch    Ticket     Fare        Cabin Embarked  
1        0  PC 17599  71.2833          C85        C  
3        0    113803  53.1000         C123        S  
6        0     17463  51.8625          E46        S  
11       0    113783  26.5500         C103        S  
23       0    113788  35.5000           A6        S  
..     ...       ...      ...          ...      ...  
871      1     11751  52.5542          D35        S  
872      0       695   5.0000  B51 B53 B55        S  
879      1     11767  83.1583          C50        C  
887      0    112053  30.0000          B42        S  
889      0    111369  30.0000         C148        C  

[216 rows x 12 columns]

In [197]:
891-216

675

675 is number of the Titanic passengers from cabin class 2 and 3.

In [199]:
df[(df['Pclass'] > 1)]

PassengerId  Survived  Pclass                                      Name  \
0              1         0       3                   Braund, Mr. Owen Harris   
2              3         1       3                    Heikkinen, Miss. Laina   
4              5         0       3                  Allen, Mr. William Henry   
5              6         0       3                          Moran, Mr. James   
7              8         0       3            Palsson, Master. Gosta Leonard   
..           ...       ...     ...                                       ...   
884          885         0       3                    Sutehall, Mr. Henry Jr   
885          886         0       3      Rice, Mrs. William (Margaret Norton)   
886          887         0       2                     Montvila, Rev. Juozas   
888          889         0       3  Johnston, Miss. Catherine Helen "Carrie"   
890          891         0       3                       Dooley, Mr. Patrick   

        Sex   Age  SibSp  Parch            Ticket     Fare Cabin Embarked  
0      male  22.0      1      0         A/5 21171   7.2500   NaN        S  
2    female  26.0      0      0  STON/O2. 3101282   7.9250   NaN        S  
4      male  35.0      0      0            373450   8.0500   NaN        S  
5      male   NaN      0      0            330877   8.4583   NaN        Q  
7      male   2.0      3      1            349909  21.0750   NaN        S  
..      ...   ...    ...    ...               ...      ...   ...      ...  
884    male  25.0      0      0   SOTON/OQ 392076   7.0500   NaN        S  
885  female  39.0      0      5            382652  29.1250   NaN        Q  
886    male  27.0      0      0            211536  13.0000   NaN        S  
888  female   NaN      1      2        W./C. 6607  23.4500   NaN        S  
890    male  32.0      0      0            370376   7.7500   NaN        Q  

[675 rows x 12 columns]

In [200]:
df[df["Pclass"].isin([2, 3])]

PassengerId  Survived  Pclass                                      Name  \
0              1         0       3                   Braund, Mr. Owen Harris   
2              3         1       3                    Heikkinen, Miss. Laina   
4              5         0       3                  Allen, Mr. William Henry   
5              6         0       3                          Moran, Mr. James   
7              8         0       3            Palsson, Master. Gosta Leonard   
..           ...       ...     ...                                       ...   
884          885         0       3                    Sutehall, Mr. Henry Jr   
885          886         0       3      Rice, Mrs. William (Margaret Norton)   
886          887         0       2                     Montvila, Rev. Juozas   
888          889         0       3  Johnston, Miss. Catherine Helen "Carrie"   
890          891         0       3                       Dooley, Mr. Patrick   

        Sex   Age  SibSp  Parch            Ticket     Fare Cabin Embarked  
0      male  22.0      1      0         A/5 21171   7.2500   NaN        S  
2    female  26.0      0      0  STON/O2. 3101282   7.9250   NaN        S  
4      male  35.0      0      0            373450   8.0500   NaN        S  
5      male   NaN      0      0            330877   8.4583   NaN        Q  
7      male   2.0      3      1            349909  21.0750   NaN        S  
..      ...   ...    ...    ...               ...      ...   ...      ...  
884    male  25.0      0      0   SOTON/OQ 392076   7.0500   NaN        S  
885  female  39.0      0      5            382652  29.1250   NaN        Q  
886    male  27.0      0      0            211536  13.0000   NaN        S  
888  female   NaN      1      2        W./C. 6607  23.4500   NaN        S  
890    male  32.0      0      0            370376   7.7500   NaN        Q  

[675 rows x 12 columns]

I want to work with passenger data for which the age is known.

In [206]:
df.dropna(subset=['Age'])

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
885          886         0       3   
886          887         0       2   
887          888         1       1   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
885               Rice, Mrs. William (Margaret Norton)  female  39.0      0   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
885      5            382652  29.1250   NaN        Q  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[714 rows x 12 columns]

In [203]:
df = df[df['Age'].notna()]

In [205]:
df

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
885          886         0       3   
886          887         0       2   
887          888         1       1   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
885               Rice, Mrs. William (Margaret Norton)  female  39.0      0   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
885      5            382652  29.1250   NaN        Q  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[714 rows x 12 columns]

**I’m interested in the names of the passengers older than 35 years.**

In [209]:
df[df['Age']>35]

PassengerId  Survived  Pclass  \
1              2         1       1   
6              7         0       1   
11            12         1       1   
13            14         0       3   
15            16         1       2   
..           ...       ...     ...   
865          866         1       2   
871          872         1       1   
873          874         0       3   
879          880         1       1   
885          886         0       3   

                                                  Name     Sex   Age  SibSp  \
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
6                              McCarthy, Mr. Timothy J    male  54.0      0   
11                            Bonnell, Miss. Elizabeth  female  58.0      0   
13                         Andersson, Mr. Anders Johan    male  39.0      1   
15                    Hewlett, Mrs. (Mary D Kingcome)   female  55.0      0   
..                                                 ...     ...   ...    ...   
865                           Bystrom, Mrs. (Karolina)  female  42.0      0   
871   Beckwith, Mrs. Richard Leonard (Sallie Monypeny)  female  47.0      1   
873                        Vander Cruyssen, Mr. Victor    male  47.0      0   
879      Potter, Mrs. Thomas Jr (Lily Alexenia Wilson)  female  56.0      0   
885               Rice, Mrs. William (Margaret Norton)  female  39.0      0   

     Parch    Ticket     Fare Cabin Embarked  
1        0  PC 17599  71.2833   C85        C  
6        0     17463  51.8625   E46        S  
11       0    113783  26.5500  C103        S  
13       5    347082  31.2750   NaN        S  
15       0    248706  16.0000   NaN        S  
..     ...       ...      ...   ...      ...  
865      0    236852  13.0000   NaN        S  
871      1     11751  52.5542   D35        S  
873      0    345765   9.0000   NaN        S  
879      1     11767  83.1583   C50        C  
885      5    382652  29.1250   NaN        Q  

[217 rows x 12 columns]

**I’m interested in rows 10 till 25 and columns 3 to 5**

In [235]:
df_drop_c=df.copy() # Copy to avoid modifying the original data
Select_column=column[0:2] # Select columns that have missing #columns described above # look at the non-null columns above
Select_column1= column[6:]
df_drop_c.drop(Select_column, axis=1, inplace=True)
df_drop_c2=df_drop_c.copy()
df_drop_c2.drop(Select_column1, axis=1, inplace=True)

df_drop_c2.drop(df.index[[-2]], inplace=True)
df_drop_c2
df_drop_c2[-2:]



Pclass                          Name     Sex   Age
887       1  Graham, Miss. Margaret Edith  female  19.0
890       3           Dooley, Mr. Patrick    male  32.0

In [236]:
df.iloc[9:25, 2:5]

Pclass                                               Name     Sex
10       3                    Sandstrom, Miss. Marguerite Rut  female
11       1                           Bonnell, Miss. Elizabeth  female
12       3                     Saundercock, Mr. William Henry    male
13       3                        Andersson, Mr. Anders Johan    male
14       3               Vestrom, Miss. Hulda Amanda Adolfina  female
15       2                   Hewlett, Mrs. (Mary D Kingcome)   female
16       3                               Rice, Master. Eugene    male
18       3  Vander Planke, Mrs. Julius (Emelia Maria Vande...  female
20       2                               Fynney, Mr. Joseph J    male
21       2                              Beesley, Mr. Lawrence    male
22       3                        McGowan, Miss. Anna "Annie"  female
23       1                       Sloper, Mr. William Thompson    male
24       3                      Palsson, Miss. Torborg Danira  female
25       3  Asplund, Mrs. Carl Oscar (Selma Augusta Emilia...  female
27       1                     Fortune, Mr. Charles Alexander    male
30       1                           Uruchurtu, Don. Manuel E    male

**Assign the name anonymous to the first 3 elements of the third column**

In [246]:
df

PassengerId   Survived     Pclass  \
0     anormnous  anormnous  anormnous   
1     anormnous  anormnous  anormnous   
2     anormnous  anormnous  anormnous   
3             4          1          1   
4             5          0          3   
..          ...        ...        ...   
885         886          0          3   
886         887          0          2   
887         888          1          1   
889         890          1          1   
890         891          0          3   

                                             Name        Sex        Age  \
0                                       anormnous  anormnous  anormnous   
1                                       anormnous  anormnous  anormnous   
2                                       anormnous  anormnous  anormnous   
3    Futrelle, Mrs. Jacques Heath (Lily May Peel)     female         35   
4                        Allen, Mr. William Henry       male         35   
..                                            ...        ...        ...   
885          Rice, Mrs. William (Margaret Norton)     female         39   
886                         Montvila, Rev. Juozas       male         27   
887                  Graham, Miss. Margaret Edith     female         19   
889                         Behr, Mr. Karl Howell       male         26   
890                           Dooley, Mr. Patrick       male         32   

         SibSp      Parch     Ticket       Fare      Cabin   Embarked  
0    anormnous  anormnous  anormnous  anormnous  anormnous  anormnous  
1    anormnous  anormnous  anormnous  anormnous  anormnous  anormnous  
2    anormnous  anormnous  anormnous  anormnous  anormnous  anormnous  
3            1          0     113803       53.1       C123          S  
4            0          0     373450       8.05        NaN          S  
..         ...        ...        ...        ...        ...        ...  
885          0          5     382652     29.125        NaN          Q  
886          0          0     211536         13        NaN          S  
887          0          0     112053         30        B42          S  
889          0          0     111369         30       C148          C  
890          0          0     370376       7.75        NaN          Q  

[714 rows x 12 columns]

In [247]:
k=df['Name']

In [248]:
k[0:3]='dd'

In [249]:
k

0                                                dd
1                                                dd
2                                                dd
3      Futrelle, Mrs. Jacques Heath (Lily May Peel)
4                          Allen, Mr. William Henry
                           ...                     
885            Rice, Mrs. William (Margaret Norton)
886                           Montvila, Rev. Juozas
887                    Graham, Miss. Margaret Edith
889                           Behr, Mr. Karl Howell
890                             Dooley, Mr. Patrick
Name: Name, Length: 714, dtype: object

In [243]:
df.iloc[3, 0:3]

PassengerId    4
Survived       1
Pclass         1
Name: 3, dtype: object

**Number of unique passanger**

In [9]:
df['PassengerId'].unique()

array([  1,   2,   3,   4,   5,   6,   7,   8,   9,  10,  11,  12,  13,
        14,  15,  16,  17,  18,  19,  20,  21,  22,  23,  24,  25,  26,
        27,  28,  29,  30,  31,  32,  33,  34,  35,  36,  37,  38,  39,
        40,  41,  42,  43,  44,  45,  46,  47,  48,  49,  50,  51,  52,
        53,  54,  55,  56,  57,  58,  59,  60,  61,  62,  63,  64,  65,
        66,  67,  68,  69,  70,  71,  72,  73,  74,  75,  76,  77,  78,
        79,  80,  81,  82,  83,  84,  85,  86,  87,  88,  89,  90,  91,
        92,  93,  94,  95,  96,  97,  98,  99, 100, 101, 102, 103, 104,
       105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117,
       118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130,
       131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143,
       144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156,
       157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 169,
       170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 180, 18

In [10]:
df['PassengerId'].count()

891

In [ ]:
Total number of passangers are 891.

**Number of unique passanger that survived**

In [21]:
df[df['Survived'] ==1]

PassengerId  Survived  Pclass  \
1              2         1       1   
2              3         1       3   
3              4         1       1   
8              9         1       3   
9             10         1       2   
..           ...       ...     ...   
875          876         1       3   
879          880         1       1   
880          881         1       2   
887          888         1       1   
889          890         1       1   

                                                  Name     Sex   Age  SibSp  \
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
8    Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                  Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   
..                                                 ...     ...   ...    ...   
875                   Najib, Miss. Adele Kiamie "Jane"  female  15.0      0   
879      Potter, Mrs. Thomas Jr (Lily Alexenia Wilson)  female  56.0      0   
880       Shelley, Mrs. William (Imanita Parrish Hall)  female  25.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
889                              Behr, Mr. Karl Howell    male  26.0      0   

     Parch            Ticket     Fare Cabin Embarked  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
8        2            347742  11.1333   NaN        S  
9        0            237736  30.0708   NaN        C  
..     ...               ...      ...   ...      ...  
875      0              2667   7.2250   NaN        C  
879      1             11767  83.1583   C50        C  
880      1            230433  26.0000   NaN        S  
887      0            112053  30.0000   B42        S  
889      0            111369  30.0000  C148        C  

[342 rows x 12 columns]

In [22]:
df[df['Survived'] ==1].count()

PassengerId    342
Survived       342
Pclass         342
Name           342
Sex            342
Age            290
SibSp          342
Parch          342
Ticket         342
Fare           342
Cabin          136
Embarked       340
dtype: int64

In [ ]:
342 passangers survived

In [ ]:
**Number of passanger that survived were females** 

In [23]:
number_of_passangers_that_survived =df[df['Survived'] ==1]

In [24]:
number_of_passangers_that_survived

PassengerId  Survived  Pclass  \
1              2         1       1   
2              3         1       3   
3              4         1       1   
8              9         1       3   
9             10         1       2   
..           ...       ...     ...   
875          876         1       3   
879          880         1       1   
880          881         1       2   
887          888         1       1   
889          890         1       1   

                                                  Name     Sex   Age  SibSp  \
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
8    Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                  Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   
..                                                 ...     ...   ...    ...   
875                   Najib, Miss. Adele Kiamie "Jane"  female  15.0      0   
879      Potter, Mrs. Thomas Jr (Lily Alexenia Wilson)  female  56.0      0   
880       Shelley, Mrs. William (Imanita Parrish Hall)  female  25.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
889                              Behr, Mr. Karl Howell    male  26.0      0   

     Parch            Ticket     Fare Cabin Embarked  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
8        2            347742  11.1333   NaN        S  
9        0            237736  30.0708   NaN        C  
..     ...               ...      ...   ...      ...  
875      0              2667   7.2250   NaN        C  
879      1             11767  83.1583   C50        C  
880      1            230433  26.0000   NaN        S  
887      0            112053  30.0000   B42        S  
889      0            111369  30.0000  C148        C  

[342 rows x 12 columns]

In [29]:
df[(df['Survived'] ==1) & (df['Sex'] =='female')]

PassengerId  Survived  Pclass  \
1              2         1       1   
2              3         1       3   
3              4         1       1   
8              9         1       3   
9             10         1       2   
..           ...       ...     ...   
874          875         1       2   
875          876         1       3   
879          880         1       1   
880          881         1       2   
887          888         1       1   

                                                  Name     Sex   Age  SibSp  \
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
8    Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                  Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   
..                                                 ...     ...   ...    ...   
874              Abelson, Mrs. Samuel (Hannah Wizosky)  female  28.0      1   
875                   Najib, Miss. Adele Kiamie "Jane"  female  15.0      0   
879      Potter, Mrs. Thomas Jr (Lily Alexenia Wilson)  female  56.0      0   
880       Shelley, Mrs. William (Imanita Parrish Hall)  female  25.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   

     Parch            Ticket     Fare Cabin Embarked  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
8        2            347742  11.1333   NaN        S  
9        0            237736  30.0708   NaN        C  
..     ...               ...      ...   ...      ...  
874      0         P/PP 3381  24.0000   NaN        C  
875      0              2667   7.2250   NaN        C  
879      1             11767  83.1583   C50        C  
880      1            230433  26.0000   NaN        S  
887      0            112053  30.0000   B42        S  

[233 rows x 12 columns]

In [ ]:
342 of the survived passangers are females

**What is the highest fare paid ?**

In [30]:
df['Fare'].max()

512.3292

What is the lowest fare paid ?

In [31]:
df['Fare'].min()

0.0

What is the lowest fare paid that is not equal to 0?

In [33]:
df[df['Fare'] != 0].min()

PassengerId                      1
Survived                         0
Pclass                           1
Name           Abbing, Mr. Anthony
Sex                         female
Age                           0.42
SibSp                            0
Parch                            0
Ticket                      110152
Fare                        4.0125
dtype: object

In [50]:
df.Fare.nsmallest(16).iloc[-1]

4.0125

In [ ]:
4.0125 is the lowest fair paid

What is the median fare paid ?

In [58]:
df['Fare'].median()

14.4542

How many people paid more than the median fare developed above?

In [65]:
df[df['Fare']>14.4542].count()

PassengerId    444
Survived       444
Pclass         444
Name           444
Sex            444
Age            371
SibSp          444
Parch          444
Ticket         444
Fare           444
Cabin          180
Embarked       442
dtype: int64

In [68]:
df[df['Fare']>df['Fare'].median()].count()

PassengerId    444
Survived       444
Pclass         444
Name           444
Sex            444
Age            371
SibSp          444
Parch          444
Ticket         444
Fare           444
Cabin          180
Embarked       442
dtype: int64

444 people paid more than mean amount

What is the mean ticket fare price for each of the sex and cabin class combinations?

In [64]:
df.groupby(['Sex','Pclass'])['Fare'].mean()

Sex     Pclass
female  1         107.946275
        2          21.951070
        3          15.875369
male    1          71.142781
        2          21.113131
        3          12.162695
Name: Fare, dtype: float64

What is the number of passengers in each of the cabin classes?

In [69]:
df['Pclass'].value_counts()

3    355
1    186
2    173
Name: Pclass, dtype: int64

In [71]:
df.groupby('Pclass').count()

PassengerId  Survived  Name  Sex  Age  SibSp  Parch  Ticket  Fare  \
Pclass                                                                      
1               186       186   186  186  186    186    186     186   186   
2               173       173   173  173  173    173    173     173   173   
3               355       355   355  355  355    355    355     355   355   

        Cabin  Embarked  Names  
Pclass                          
1         160       184    186  
2          15       173    173  
3          10       355    355

Make all name characters lowercase

In [74]:
df["Name"].str.lower()

0                                braund, mr. owen harris
1      cumings, mrs. john bradley (florence briggs th...
2                                 heikkinen, miss. laina
3           futrelle, mrs. jacques heath (lily may peel)
4                               allen, mr. william henry
                             ...                        
885                 rice, mrs. william (margaret norton)
886                                montvila, rev. juozas
887                         graham, miss. margaret edith
889                                behr, mr. karl howell
890                                  dooley, mr. patrick
Name: Name, Length: 714, dtype: object